/*
*  This file is part of aasdk library project.
*  Copyright (C) 2018 f1x.studio (Michal Szwaj)
*
*  aasdk is free software: you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 3 of the License, or
*  (at your option) any later version.

*  aasdk is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  You should have received a copy of the GNU General Public License
*  along with aasdk. If not, see <http://www.gnu.org/licenses/>.
*/

syntax="proto2";

package oaa.proto.messages;

message ServiceDiscoveryRequest
{
    // Fields 1-3 discovered via APK proto decoder (aahi.java, AA v16.1)
    // Confirmed as PNG images via live capture (Samsung S25 Ultra, 2026-02-24):
    //   field 1: 32x32 PNG icon, field 2: 64x64 PNG icon, field 3: 128x128 PNG icon
    optional bytes phone_icon_small = 1;   // 32x32 PNG — phone/AA app icon
    optional bytes phone_icon_medium = 2;  // 64x64 PNG — phone/AA app icon
    optional bytes phone_icon_large = 3;   // 128x128 PNG — phone/AA app icon

    required string device_name = 4;
    required string device_brand = 5;

    // Field 6: sub-message with UUID string at sub-field 1 (seen in live capture)
    // e.g. { 1: "13504ec4-0e63-4e0f-a4d6-38472b4c39f7" }
    // Likely session/device UUID for reconnection tracking
    optional bytes session_info = 6;
}
